File: tests/fixtures/ts-project/src/payment-service.ts
export class PaymentService {
  async charge(customerId: string, amount: number): Promise<void> {
    console.log(`Charging ${customerId} $${amount}`);
  }

  async refund(transactionId: string): Promise<void> {
    console.log(`Refunding ${transactionId}`);
  }
}

File: tests/fixtures/java-project/src/main/java/com/example/payment/PaymentService.java
package com.example.payment;

public class PaymentService {

    private final PaymentGateway paymentGateway;

    public PaymentService(PaymentGateway paymentGateway) {
        this.paymentGateway = paymentGateway;
    }

    public void charge(PaymentInfo paymentInfo) {
        paymentGateway.processPayment(paymentInfo.getAmount());
    }

    public void refund(String transactionId) {
        paymentGateway.processRefund(transactionId);
    }
}
